My Project/clsGenerateTranslations.vb

```diff
@@ -0,0 +1,558 @@
+''' In order to enforce this (and prevent developers from using this class in an unintended way), 
+''' the class is declared as 'NotInheritable` and the constructor is declared as 'Private'.</para>
+''' </summary>
+'''------------------------------------------------------------------------------------------------
+Public NotInheritable Class clsGenerateTranslations
+
+    '''--------------------------------------------------------------------------------------------
+    ''' <summary> 
+    ''' Declare constructor 'Private' to prevent instantiation of this class (see class comments 
+    ''' for more details). 
+    ''' </summary>
+    '''--------------------------------------------------------------------------------------------
+    Private Sub New()
+    End Sub
+
+
+    '''--------------------------------------------------------------------------------------------
+    ''' <summary>
+    ''' Updates the 'form_controls' database table with controls in the passed datatable
+    ''' </summary>
+    ''' <param name="strDataSource">The file path to the sqlite database file</param>
+    ''' <param name="datatableControls">The form controls datatable with 3 columns; form_name, control_name, id_text.</param>
+    ''' <param name="strTranslateIgnoreFilePath">Optional. The file path to the translate ignore file. If passed translate ignore controls will also be processed.</param>
+    ''' <returns>The number of successful updates.</returns>
+    '''--------------------------------------------------------------------------------------------
+    Public Shared Function UpdateFormControlsTable(strDataSource As String, datatableControls As DataTable, Optional strTranslateIgnoreFilePath As String = "") As Integer
+        Dim iRowsUpdated As Integer = 0
+        Try
+            'connect to the database and execute the SQL command 
+            Dim clsBuilder As New SQLiteConnectionStringBuilder With {
+                    .FailIfMissing = True,
+                    .DataSource = strDataSource}
+            Using clsConnection As New SQLiteConnection(clsBuilder.ConnectionString)
+                clsConnection.Open()
+                'todo. do batch execution for optimal performance
+                For Each row As DataRow In datatableControls.Rows
+
+                    Dim paramFormName As New SQLiteParameter("form_name", row.Field(Of String)(0))
+                    Dim paramControlName As New SQLiteParameter("control_name", row.Field(Of String)(1))
+                    Dim paramIdText As New SQLiteParameter("id_text", row.Field(Of String)(2))
+
+                    'delete record if exists first 
+                    Dim sqlDeleteCommand As String = "DELETE FROM form_controls WHERE form_name = @form_name AND control_name=@control_name"
+                    Using cmdDelete As New SQLiteCommand(sqlDeleteCommand, clsConnection)
+                        cmdDelete.Parameters.Add(paramFormName)
```

Comment:
In `clsTranslateWinForms`, parameters are added with code like `cmdSelect.Parameters.Add(New SQLiteParameter("id_text", strIdText))`.
What's the reason for doing it differently here?